operator/validator/controller.go

```diff
@@ -123,16 +126,43 @@ type nonCommitteeValidator struct {
 	sync.Mutex
 }
 
+type Nonce uint16
+
+type Recipients interface {
+	GetRecipientData(r basedb.Reader, owner common.Address) (*registrystorage.RecipientData, bool, error)
+}
+
+type SharesStorage interface {
+	Get(txn basedb.Reader, pubKey []byte) *types.SSVShare
+	List(txn basedb.Reader, filters ...registrystorage.SharesFilter) []*types.SSVShare
+	UpdateValidatorMetadata(pk string, metadata *beaconprotocol.ValidatorMetadata) error
+}
+
+type P2PNetwork interface {
+	Close() error
+	Setup(logger *zap.Logger) error
+	Start(logger *zap.Logger) error
+	UpdateSubnets(logger *zap.Logger)
+	SubscribeAll(logger *zap.Logger) error
+	Broadcast(message *spectypes.SSVMessage) error
+	Subscribe(vpk spectypes.ValidatorPK) error
+	UseMessageRouter(router network.MessageRouter)
+	Peers(pk spectypes.ValidatorPK) ([]peer.ID, error)
+	SubscribeRandoms(logger *zap.Logger, numSubnets int) error
+	RegisterHandlers(logger *zap.Logger, handlers ...*p2pprotocol.SyncHandler)
+	ReportValidation(logger *zap.Logger, message *spectypes.SSVMessage, res p2pprotocol.MsgValidationResult)
```

Comment:
are all of these really used by validatorController or by where it passes it to. if not reduce the interface as much as possible